packages/edit-site/src/components/save-hub/index.js

```diff
@@ -32,29 +55,106 @@ export default function SaveHub() {
 		[]
 	);
 
+	const {
+		editEntityRecord,
+		saveEditedEntityRecord,
+		__experimentalSaveSpecifiedEntityEdits: saveSpecifiedEntityEdits,
+	} = useDispatch( coreStore );
+
 	const disabled = isSaving || ( ! isDirty && ! isPreviewingTheme() );
 
+	// if we have only one unsaved change and it matches current context, we can show a more specific label
+	let dirtyLocal = null;
+
+	if ( countUnsavedChanges === 1 ) {
+		// if we are on global styles
+		if ( params.path?.includes( 'wp_global_styles' ) ) {
+			dirtyLocal = dirtyRecords.find(
+				( record ) => record.name === 'globalStyles'
+			);
+		}
+		// if we are on pages
+		else if ( !! params.postType ) {
+			dirtyLocal = dirtyRecords.find(
+				( record ) =>
+					record.name === params.postType &&
+					String( record.key ) === params.postId
+			);
+		}
+	}
+
+	let label = dirtyLocal
+		? __( 'Save' )
+		: sprintf(
+				// translators: %d: number of unsaved changes (number).
+				_n(
+					'Review %d change…',
+					'Review %d changes…',
+					countUnsavedChanges
+				),
+				countUnsavedChanges
+		  );
+
+	if ( isSaving ) {
+		label = __( 'Saving' );
+	}
+
+	const saveCurrentEntity = async () => {
+		if ( ! dirtyLocal ) return;
+
+		const { kind, name, key, property } = dirtyLocal;
+
+		try {
+			if ( 'root' === dirtyLocal.kind && 'site' === name ) {
+				await saveSpecifiedEntityEdits( 'root', 'site', undefined, [
+					property,
+				] );
+			} else {
+				if (
+					PUBLISH_ON_SAVE_ENTITIES.some(
+						( typeToPublish ) =>
+							typeToPublish.kind === kind &&
+							typeToPublish.name === name
+					)
+				) {
+					editEntityRecord( kind, name, key, { status: 'publish' } );
+				}
+
+				await saveEditedEntityRecord( kind, name, key );
+			}
+
+			__unstableMarkLastChangeAsPersistent();
+
+			createSuccessNotice( __( 'Site updated.' ), {
+				type: 'snackbar',
+			} );
+		} catch ( error ) {
+			createErrorNotice( `${ __( 'Saving failed.' ) } ${ error }` );
+		}
+	};
+
 	return (
 		<HStack className="edit-site-save-hub" alignment="right" spacing={ 4 }>
-			{ isDirty && (
-				<span>
-					{ sprintf(
-						// translators: %d: number of unsaved changes (number).
-						_n(
-							'%d unsaved change',
-							'%d unsaved changes',
-							countUnsavedChanges
-						),
-						countUnsavedChanges
-					) }
-				</span>
+			{ dirtyLocal ? (
```

Comment:
Why does this need to render a different button than the "Save button"? Do we want to change just the label? Is it something else?

Reply:
@youknowriad I was just wary of adding too much complexity to the save button that's shared with the editor/edit mode. Thinking was that separating the two behaviours (review modal vs directly saving) would be less risky.